src/main/scala/com/google/cloud/spark/bigquery/direct/DirectBigQueryRelation.scala

```diff
@@ -127,23 +150,97 @@ private[bigquery] class DirectBigQueryRelation(
         prunedSchema,
         requiredColumns,
         options,
-        getClient).asInstanceOf[RDD[Row]]
+        getClient,
+        bigQueryClient).asInstanceOf[RDD[Row]]
 
     } finally {
       // scanTable returns immediately not after the actual data is read.
       client.close()
     }
   }
 
+  def getActualTable(
+      requiredColumns: Array[String],
+      filters: Array[Filter]
+    ): TableInfo = {
+    val tableDefinition = table.getDefinition[TableDefinition]
+    val tableType = tableDefinition.getType
+    if(options.viewsEnabled && TableDefinition.Type.VIEW == tableType) {
+      // get it from the view
+      val querySql = createSql(tableDefinition.getSchema, requiredColumns, filters)
+      log.debug(s"querySql is $querySql")
+      destinationTableCache.get(querySql, DestinationTableBuilder(querySql))
+    } else {
+      // use the default one
+      table
+    }
+  }
+
+  def createTableFromQuery(querySql: String): TableInfo = {
+    val destinationTable = createDestinationTable
+    log.debug(s"destinationTable is $destinationTable")
+    val jobInfo = JobInfo.of(
+      QueryJobConfiguration
+        .newBuilder(querySql)
+        .setDestinationTable(destinationTable)
+        .build())
+    log.debug(s"running query $jobInfo")
+    val job = bigQuery.create(jobInfo).waitFor()
+    log.debug(s"job has finished. $job")
+    if(job.getStatus.getError != null) {
+      BigQueryUtil.convertAndThrow(job.getStatus.getError)
+    }
+    // add expiration time to the table
+    val createdTable = bigQuery.getTable(destinationTable)
+    val expirationTime = createdTable.getCreationTime +
+      TimeUnit.HOURS.toMillis(options.viewExpirationTimeInHours)
+    val updatedTable = bigQuery.update(createdTable.toBuilder
+      .setExpirationTime(expirationTime)
+      .build())
+    updatedTable
+  }
+
+  def createSql(schema: Schema, requiredColumns: Array[String], filters: Array[Filter]): String = {
+    val columns = if(requiredColumns.isEmpty) {
+      val sparkSchema = SchemaConverters.toSpark(schema)
+      sparkSchema.map(f => s"`${f.name}`").mkString(",")
+    } else {
+      requiredColumns.map(c => s"`$c`").mkString(",")
+    }
+
+    val whereClause = createWhereClause(filters)
+      .map(f => s"WHERE $f")
+      .getOrElse("")
+
+    return s"SELECT $columns FROM `$tableName` $whereClause"
+  }
+
+  // return empty if no filters are used
+  def createWhereClause(filters: Array[Filter]): Option[String] = {
+    val filtersString = DirectBigQueryRelation.compileFilters(filters)
+    BigQueryUtil.noneIfEmpty(filtersString)
+  }
+
+  def createDestinationTable: TableId = {
+    val uuid = UUID.randomUUID()
+    val name = s"_sbc_${uuid.getMostSignificantBits.toHexString}${uuid.getLeastSignificantBits.toHexString}"
+    TableId.of(tableId.getProject, tableId.getDataset, name)
```

Comment:
You might consider making this configurable -- in some cases, it won't be appropriate to put the temporary table in the same dataset as the source table. Beam, for example, creates not only a temporary table, but a temporary dataset as well which it deletes at the end of pipeline execution.

Reply:
Added the configuration options, see https://github.com/GoogleCloudPlatform/spark-bigquery-connector/pull/83/commits/1cb67b5f60c38fd9575050b53ffd7d784dfc3b50